### ISSUE TITLE
getMetadata for directory does not return directory metadata

### ISSUE THREAD

**Issue by spwilson806** [Potherca/flysystem-github#6]
When calling getMetadata on a directory entity, it seems to return a multidimensional array of metadata for the entities of that directory, not the metadata for the directory itself.  This seems to be inconsistent with the other native flysystem adapters that I have worked with.

**Comment by Potherca** [comment]
Thank you for reporting this issue.

This behaviour does indeed seem incorrect and not in line with expected behaviour. I am marking this as a bug.

This behaviour exists because the response from the underlying library that this adapter uses is passed on ad verbatim.

Looking at [the Github API docs on this subject](https://developer.github.com/v3/repos/contents/#response-if-content-is-a-directory), it states:

> **Response if content is a directory**
> 
> The response will be an array of objects, one object for each item in the directory.

Do you have timeframe for how soon you need/want this to be fixed?

**Comment by spwilson806** [comment]
Not a huge hurry.  I think I was hoping to use this in something that I would start demoing sometime in January.  I'm also happy to jump in and help as needed.

**Comment by spwilson806** [comment]
Just checking in to see if you had any more thoughts here or had some sort of timeframe.  Thanks!

**Comment by Potherca** [comment]
Thanks for the bump! 

I _had_ planned to do some work on this right after the holidays but unfortunately things didn't go as planned. 

I am currently in the process of looking for a new job for after my current contract ends. Any work I get to do on this project beforehand is a bonus but in all likelihood it'll have to wait until after I've landed a new contract.

**Comment by Potherca** [comment]
New contract is all but signed, so time for an update!

Work on this issue is underway. I've fixed the original issue but I've found some other discrepancies that I would also like to fix. Most of these are done in the linked pull-request.

The last thing that is needed before I can merge the changes and create a new release is refactor the unit-tests for the Api class. They have become too convoluted/complex (specifically in regards to mocks). Once that is done and all tests are green again, all is done.
